src/adapter/src/session.rs

```diff
@@ -603,11 +603,14 @@ pub type RowBatchStream = UnboundedReceiver<PeekResponseUnary>;
 pub enum TransactionStatus<T> {
     /// Idle. Matches `TBLOCK_DEFAULT`.
     Default,
-    /// Running a possibly single-query transaction. Matches
-    /// `TBLOCK_STARTED`. WARNING: This might not actually be
-    /// a single statement due to the extended protocol. Thus,
-    /// we should not perform optimizations based on this.
-    /// See: <https://git.postgresql.org/gitweb/?p=postgresql.git&a=commitdiff&h=f92944137>.
+    /// Running a single-query transaction. Matches
+    /// `TBLOCK_STARTED`. In PostgreSQL, when using the extended query protocol, this
+    /// may be upgraded into multi-statement implicit query (see [`Self::InTransactionImplicit`]).
```

Comment:
How does the upgrade into implicit multi statement happen? This seems at odds with the sentence below that says "we can guarantee that this state will always be a single-query transaction".

Reply:
I was trying to highlight that in Postgres an upgrade is possible, but in Materialize an upgrade isn't possible. I've added a sentence at the end to hopefully clarify this. Does that help enough do you think?

Reply:
Yes, thanks.